src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -91,6 +91,80 @@ public class TestExtensions
     /// </summary>
     internal bool AreDataCollectorsCached { get; set; }
 
+    /// <summary>
+    /// Merge two extension dictionaries.
+    /// </summary>
+    /// 
+    /// <param name="first">First extension dictionary.</param>
+    /// <param name="second">Second extension dictionary.</param>
+    /// 
+    /// <returns>
+    /// A dictionary representing the merger between the two input dictionaries.
+    /// </returns>
+    internal static Dictionary<string, HashSet<string>> MergeDictionaries(
+        Dictionary<string, HashSet<string>> first,
+        Dictionary<string, HashSet<string>> second)
+    {
+        var isFirstNullOrEmpty = first == null || first.Count == 0;
+        var isSecondNullOrEmpty = second == null || second.Count == 0;
+
+        // Sanity checks.
+        if (isFirstNullOrEmpty && isSecondNullOrEmpty)
+        {
+            return new Dictionary<string, HashSet<string>>();
+        }
+        if (isFirstNullOrEmpty)
+        {
+            return new Dictionary<string, HashSet<string>>(second);
+        }
+        if (isSecondNullOrEmpty)
+        {
+            return new Dictionary<string, HashSet<string>>(first);
+        }
+
+        // Copy all the keys in the first dictionary into the resulting dictionary.
+        var resultMap = new Dictionary<string, HashSet<string>>(first);
+
+        foreach (var kvp in second)
```

Comment:
[minor] You could also deconstruct the tuple here to have more meaningful "key" and "value" names.

Reply:
I'd rather keep it as it is since we decided the code is quite generic. We probably shouldn't use more specific names in this case.